R/dfm-methods.R

```diff
@@ -144,13 +144,13 @@ matrix2dfm <- function(x, docvars = NULL, meta = NULL) {
     
     if (nrow(x) == 0 && ncol(x) == 0) {
         # avoid coercion to ldiMatrix
-        x <- as(matrix(nrow = 0, ncol = 0), "dgeMatrix")
+        x <- as(as(as(matrix(nrow = 0, ncol = 0), "dMatrix"), "generalMatrix"), "unpackedMatrix")
     } else {
         x <- Matrix(x, sparse = TRUE)
```

Comment:
It might be safe enough if we ensure that `x` is a dgC or  dgT Matrix here like:
```r
x <- Matrix(x, sparse = TRUE)
x <- as(as(as(x, "CsparseMatrix"), "generalMatrix"), "dMatrix")
```
Then we need only `as(x, "CsparseMatrix")` or `as(x, "TsparseMatrix")` in other functions.

Reply:
I also changed 147 to `make_null_dfm()` since we already have a function for that.